server/stream.go

```diff
@@ -404,12 +404,26 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 		}
 	}
 
-	// Check for overlapping subjects. These are not allowed for now.
+	// Check for overlapping subjects with other streams.
+	// These are not allowed for now.
 	if jsa.subjectsOverlap(cfg.Subjects) {
 		jsa.mu.Unlock()
 		return nil, NewJSStreamSubjectOverlapError()
 	}
 
+	// Now check if we have multiple subjects they we do not overlap ourselves
+	// which would cause duplicate entries (assuming no MsgID).
+	if len(cfg.Subjects) > 1 {
+		for _, subj := range cfg.Subjects {
+			for _, tsubj := range cfg.Subjects {
+				if tsubj != subj && SubjectsCollide(tsubj, subj) {
+					jsa.mu.Unlock()
+					return nil, fmt.Errorf("subject %q overlaps with %q", subj, tsubj)
```

Comment:
2 things:
- is there a formal NewJSxx error already for that or not?
- more importantly, if a stream already exists like that, does it mean that it will not be recovered? does that leave the system in a precarious condition?

I would recommend make a manual test and see how it behaves on an upgrade to make sure it is not too bad.

Reply:
Good point, yes if it was created it would not be able to restore.

No error for that one, we have one for colliding with another stream's subjects.

Will also add test for update stream, etc.

Reply:
ok added test for update, will do manual test on recover now.

Reply:
I think it is ok. Can always use old server to restore and snapshot and edit subjects.

<img width="913" alt="image" src="https://user-images.githubusercontent.com/90097/187800301-9a103932-5b22-4dc8-a268-2914a8874d65.png">